system/perf_res_processor.py

```diff
@@ -0,0 +1,269 @@
+                                ).decode().strip().splitlines()
+                            except subprocess.CalledProcessError:
+                                pattern_res += []
+
+                # find errors in all .xz
+                xz_names = [x for x in xz_file_names if x.__contains__(node_key)]
+                if xz_names:
+                    for xz_name in xz_names:
+                        try:
+                            res += subprocess.check_output(
+                                ['xzgrep', 'ERROR', path + xz_name]
+                            ).decode().strip().splitlines()
+                        except subprocess.CalledProcessError:
+                            res += []
+
+                        # grep patterns and add them too
+                        for item in LOG_PATTERNS:
+                            try:
+                                pattern_res += subprocess.check_output(
+                                    ['xzgrep', item, path + xz_name]
+                                ).decode().strip().splitlines()
+                            except subprocess.CalledProcessError:
+                                pattern_res += []
+
+                results.append(res)
+                pattern_results.append(pattern_res)
+
+                # create file with error entries for each node
+                with open(SUB_DIRS[i]+'error_log_entries.txt', 'w') as f:
+                    for item in res:
+                        f.write('{}\n'.format(item))
+
+                # create file with pattern matching entries for each node
+                with open(SUB_DIRS[i]+'pattern_log_entries.txt', 'w') as f:
+                    for item in pattern_res:
+                        f.write('{}\n'.format(item))
+
+            return results, pattern_results
+
+        error_results, pattern_results = get_log_errors_as_lists()
+
+        for i, item in enumerate(error_results, start=1):
+            self._report.loc[[i], ['LOG_ERRORS']] = len(item)
+        for i, item in enumerate(pattern_results, start=1):
+            self._report.loc[[i], ['PATTERN_MATCHES']] = len(item)
+
+    @staticmethod
+    def process_metrics(path_from=METRICS_DIR, paths_to=SUB_DIRS):
+        metric_file_names = sorted(
+                [x for x in os.listdir(path_from) if not x.__contains__('summary')],
+                key=NATURAL_SORTING
+            )
+
+        # create metrics figure for each node
+        assert all(
+            [plot_metrics([path_from+metric_file_name], path_to+'Figure.png') is None
+             for metric_file_name, path_to in zip(metric_file_names, paths_to)]
+        )
+
+        summary_file_names = sorted(
+                [x for x in os.listdir(path_from) if x.__contains__('summary')],
+                key=NATURAL_SORTING
+            )
+
+        # copy metrics summary for each node
+        assert all(
+            [copyfile(path_from+summary_file_name, path_to+summary_file_name) is not None
```

Comment:
i think, that os.path.join is more clearly

Reply:
Done.